specification.markdown

```diff
@@ -155,11 +155,21 @@ Class D    | 4      | 0x101FFFFF - 0x1FFFFFFE |   268,338,304
 
-An Element Data Size with all VINT_DATA bits set to one is reserved as an indicator that the size of the EBML Element is unknown. The only reserved value for the VINT_DATA of Element Data Size is all bits set to one. An EBML Element with an unknown Element Data Size is referred to as an Unknown-Sized Element. A Master Element MAY be an Unknown-Sized Element; however an EBML Element that is not a Master Element MUST NOT be an Unknown-Sized Element. Master Elements MUST NOT use an unknown size unless the unknownsizeallowed attribute of their EBML Schema is set to true (see [the section on the unknownsizeallowed attribute](#unknownsizeallowed)). The use of Unknown-Sized Elements allows for an EBML Element to be written and read before the size of the EBML Element is known. Unknown-Sized Element MUST NOT be used or defined unnecessarily; however if the Element Data Size is not known before the Element Data is written, such as in some cases of data streaming, then Unknown-Sized Elements MAY be used. The end of an Unknown-Sized Element is determined by whichever comes first: the end of the file or the beginning of the next EBML Element, defined by this document or the corresponding EBML Schema, that is not independently valid as Descendant Element of the Unknown-Sized Element.
+## Unknown Data Size
+
+An Element Data Size with all VINT_DATA bits set to one is reserved as an indicator that the size of the EBML Element is unknown. The only reserved value for the VINT_DATA of Element Data Size is all bits set to one. An EBML Element with an unknown Element Data Size is referred to as an Unknown-Sized Element. A Master Element MAY be an Unknown-Sized Element; however an EBML Element that is not a Master Element MUST NOT be an Unknown-Sized Element. Master Elements MUST NOT use an unknown size unless the unknownsizeallowed attribute of their EBML Schema is set to true (see [the section on the unknownsizeallowed attribute](#unknownsizeallowed)). The use of Unknown-Sized Elements allows for an EBML Element to be written and read before the size of the EBML Element is known. Unknown-Sized Element MUST NOT be used or defined unnecessarily; however if the Element Data Size is not known before the Element Data is written, such as in some cases of data streaming, then Unknown-Sized Elements MAY be used. The end of an Unknown-Sized Element is determined by whichever comes first:
+
+- Any valid EBML Element according to the EBML Schema that is a Parent Element of the Unknown-Sized Element.
+- Any valid EBML Element according to the EBML Schema that is not a Descendant Element of the Unknown-Sized Element. For example a Top-Level Element.
```

Comment:
I see two potential issues:

1. The first case is a special variant of the second case. Readers might wonder why we mention parent elements explicitly when the second rule covers them already.
2. Readers might wonder how global elements play into this. Maybe mention that global elements do not end an Unknown-Sized Element.

Reply:
@mbunkus counter-proposal?

Reply:
1. The second is too broad. It's only possible to have a non-descendant element if they share a parent at the same level. Or if it's a root (then no parent).
```
A>B>C>D
A>B>E     E after D is OK
A>B>F>G   G after D is not OK
A>B'       B' after D is OK
A'           A' after D is OK
[A']>B'     B' after D is not OK
```

I'll try to come up with better text and an example.

Reply:
2. OK with the global elements not ending a unknown-length element